extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/functionExtractors.scala

```diff
@@ -29,13 +30,17 @@ object FunctionExtractor {
     loadExtractorsToMap[FunctionExtractor]
   }
 
-  def buildFunctionIdentFromQualifiedName(qualifiedName: String): (String, Option[String]) = {
-    val parts: Array[String] = qualifiedName.split("\\.", 2)
-    if (parts.length == 1) {
-      (qualifiedName, None)
+  private[authz] def buildFunctionFromQualifiedName(qualifiedName: String): Function = {
+    val partsLen = if (isSparkV34OrGreater) 3 else 2
+    val parts: Array[String] = qualifiedName.split("\\.", partsLen)
```

Comment:
What's the reason for setting the limit for splitting?
Even if the version affects the split array length, we could just fetch them by corresponding index.